compose-builder/add-security.yml

```diff
@@ -71,9 +71,8 @@ services:
       - common-security.env
       - common-sec-stage-gate.env
     volumes:
-      - edgex-init:/edgex-init:ro,z
-      - /tmp/edgex/secrets/edgex-consul:/tmp/edgex/secrets/edgex-consul:ro,z
-      - /tmp/edgex/secrets/edgex-kong:/tmp/edgex/secrets/edgex-kong:ro,z
+      - edgex-init:/edgex-init:z
+      - /tmp/edgex/secrets/edgex-consul:/tmp/edgex/secrets/edgex-consul:z
```

Comment:
Why do we need write access to these volumes now?

Reply:
we save the bootstrap token into this volume

Reply:
What about edgex-init?  What do we write there?

Also, I think it would be safer to write the bootstrap token to secrets/edge-kong than give edgex-kong access to read secrets/edgex-consul volume to read the bootstrap token for purposes of setting up the reverse proxy.

Reply:
> What about edgex-init? What do we write there?

the sentinel file, edgex-init is regular volume, so we want to write there in the case of cleaning up docker's regular volume via `docker volume prune`. If we were writing it to edgex-consul, then it would not get clean up since that one is bind-mount volume to the host system, so for state-consistency  issue, we decide to write it to this regular docker volume.


> 
> Also, I think it would be safer to write the bootstrap token to secrets/edge-kong than give edgex-kong access to read secrets/edgex-consul volume to read the bootstrap token for purposes of setting up the reverse proxy. Consul is an "internal" service that is more trusted than an externally facing service.

not quite understand why it is safer in kong's volume, it has to be written volume anayway

Reply:
> the sentinel file, edgex-init is regular volume, so we want to write there in the case of cleaning up docker's regular volume via `docker volume prune`. If we were writing it to edgex-consul, then it would not get clean up since that one is bind-mount volume to the host system, so for state-consistency issue, we decide to write it to this regular docker volume.

Why not write it to /consul/config or /consul/data which is already writable and holds consul-related data, rather than make a volume that is read-only writable to hold consul-specific data?

> not quite understand why it is safer in kong's volume, it has to be written volume anayway

Giving kong access to consul's secrets volume allows kong to impersonate consul.  Giving consul access to kong's volume allows consul to impersonate kong.  Nobody (except proxy-setup) depends on kong.  Everyone depends on consul.

Reply:
i've sent you a meeting invite tomorrow to discuss bootstrapping token volume location. not sure i can comprehend all your answers

Reply:
A note for a few factors to consider for our meeting discussion:
1. sentinel file should be in regular volume so that when consul's volume is pruned, it is gone, too
2. sentinel file by default is written to `/egex-init/consul-bootstrapper` in toml file, so we could potentially make only that volume to be writable instead of the whole /edgex-init
3. the consul process itself doesn't use edgex-consul secret volume at all
4. the secret volume is not a regular docker volume, it is bind-mount type of docker volume to the host system
5. right now, `edgex-consul` secret volume is only shared between services `secretstore-setup` and `consul`
6. `/tmp/` in linux system potentially gets cleaned up on every system reboot (depending on $TMPTIME, which usually is 0). So **will it be ok to allow  consul's bootstrap token be cleaned up and gone after reboot?** (note that it is the same principle applied to all edgex secrets tokens that are written into /tmp/ dir)
5. Q: do we need the bootstrap token later?  A: Maybe, GUI needs a token to authenticate once the default_policy in consul changed to "deny".  Whether using this token or not is a different issue.

Reply:
- we can put sentinel file in `consul/config`
- we want to write this bootstrap consul token into a separate bind-mount volume specifically for kong to use
- to put this token on the new regular docker volume (volume to be shared with proxy-setup) so that it won't suffer the reboot consequence, but user may not be easy to get it
- make it easy to get it , possibly to expand the current secret-config utility to retrieve consul bootstrap token

Reply:
implemented